3-enrich/stream-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich.stream/sources/AbstractSource.scala

```diff
@@ -154,15 +157,20 @@ abstract class AbstractSource(
 
   implicit val resolver: Resolver = igluResolver
 
+  def getPiiEvent(event: EnrichedEvent): Option[String] = {
+    if (event.pii == null || event.pii.isEmpty) None
+    else Some(event.pii)
+  }
```

Comment:
isn't this just `Option(event.pii).filter(_.nonEmpty)` ?

Reply:
I think nonEmpty will not do the right thing is if the field is actually null

Reply:
`Option(null) == None`

Reply:
👍